src/dbally/audit/event_handlers/otel_event_handler.py

```diff
@@ -0,0 +1,134 @@
+            span.set_attribute("db-ally.user.collection", user_request.collection_name)
+            if self.record_inputs:
+                span.set_attribute("db-ally.user.question", user_request.question)
+
+            return span
+
+    async def event_start(self, event: Event, request_context: Span) -> Span:
+        """
+        Starts a new event in a system as a span. Uses request span as a parent.
+
+        Args:
+            event: Event to register
+            request_context: Parent span for this event
+
+        Returns:
+            span object capturing start of execution for this event
+
+        Raises:
+            ValueError: it is thrown when unknown event type is passed as argument
+        """
+        if isinstance(event, LLMEvent):
+            with self._new_child_span(request_context, "llm") as span:
+                span.set_attribute("db-ally.llm.type", event.type)
+                if self.record_inputs:
+                    span.set_attribute("db-ally.llm.prompts", json.dumps(event.prompt))
+
+            return span
+
+        if isinstance(event, SimilarityEvent):
+            with self._new_child_span(request_context, "similarity") as span:
+                span.set_attributes(
+                    {
+                        "db-ally.similarity.store": event.store,
+                        "db-ally.similarity.fetcher": event.fetcher,
+                    }
+                )
+                if self.record_inputs:
+                    span.set_attribute("db-ally.similarity.input", event.input_value)
+
+            return span
+
+        raise ValueError(f"Unsuported event: {type(event)}")
+
+    async def event_end(self, event: Optional[Event], request_context: Span, event_context: Span) -> None:
+        """
+        Finalizes execution of the event, ending a span for this event.
+
+        Args:
+            event: optional event information
+            request_context: parent span
+            event_context: event span
+        """
+
+        if isinstance(event, LLMEvent):
+            event_context.set_attribute("db-ally.llm.response-tokes", event.completion_tokens)
+            if self.record_outputs:
+                event_context.set_attribute("db-ally.llm.response", event.response)
+
+        if isinstance(event, SimilarityEvent) and self.record_outputs:
+            event_context.set_attribute("db-ally.similarity.output", event.output_value)
+
+        event_context.set_status(StatusCode.OK)
+        event_context.end()
+
+    async def request_end(self, output: RequestEnd, request_context: Span) -> None:
+        """
+        Finalizes entire request, ending the span for this request.
+
+        Args:
+            output: output generated for this request
+            request_context: span to be closed
+        """
+        if self.record_outputs:
+            request_context.set_attribute("db-ally.response", output.result.textual_response)
```

Comment:
textual_response may not be present (nl_responder is optional). 
Maybe we should add context here (generated IQL, SQL or other query)

Reply:
Sure, I will add this